cmd/scollector/collectors/database.go

```diff
@@ -0,0 +1,345 @@
+package collectors
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	_ "bosun.org/_third_party/github.com/go-sql-driver/mysql"
+	"bosun.org/cmd/scollector/conf"
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+	"bosun.org/slog"
+)
+
+type DatabaseQuery struct {
+	name        string
+	query       string
+	description string
+	hasTime     bool
+	interval    time.Duration
+	next        time.Time
+}
+
+var DatabaseVars struct {
+	vaildDatabases map[string]bool
+}
+
+func init() {
+	DatabaseVars.vaildDatabases = map[string]bool{"mysql": true}
+}
+
+func Database(c conf.Database) error {
+	return DatabaseAddCollector(c)
+}
+
+func DatabaseAddCollector(c conf.Database) error {
+	_, found := DatabaseVars.vaildDatabases[c.Type]
```

Comment:
I don't really like the DatabaseVars existing just for this check. Just start with `if c.Type != "mysql"` and we'll go from there as appropriate.